File: Hotel/src/gestorAplicacion/hoteles/Hotel.java
package gestorAplicacion.hoteles;

import gestorAplicacion.Habitaciones.*;

import java.io.Serializable;
import java.util.*;

import baseDatos.Deserializador;

public class Hotel implements Serializable {
    private List<Habitacion> habitaciones = new ArrayList<>();

    public Hotel(){
        Deserializador.deserializar(this);
    }

    public void agragarHabitacion(Habitacion habitacion){
        habitaciones.add(habitacion);
        habitaciones.add(habitacion);
    }

    public String mostrarHabitaciones(){
        String s = "";
        for (Habitacion habitacion : habitaciones) {
            s += habitacion.toString() + System.lineSeparator();
        }
        return s;
    }

    /* getters y setters */

    public List<Habitacion> getHabitaciones() {
        return habitaciones;
    }

    public void setHabitaciones(List<Habitacion> habitaciones) {
        this.habitaciones = habitaciones;
    }

    
}

File: python/src/gestor_aplicacion/hoteles/Hotel.py
import pickle
from gestor_aplicacion.hoteles.Habitacion import Habitacion
import os
from base_datos.Desearizador import deserializar
from tkinter import Tk, Menu, Toplevel, Label



class Hotel:
    def __init__(self) -> None:
        self._habitaciones:list[Habitacion] = []
        deserializar(self)


    @property
    def get_habitaciones(self):
        return self._habitaciones

    @get_habitaciones.setter
    def set_habitaciones(self, lista_habitaciones):
        self._habitaciones = lista_habitaciones

    def añadir_habitacion(self, habitacion:Habitacion) -> None:
        self.get_habitaciones.append(habitacion)

    def mostrar_habitaciones(self):
        ventana_datos= Toplevel()
        ventana_datos.geometry("400x300")
        ventana_datos.title("Habitaciones")
        for habitacion in self._habitaciones:
            label = Label(ventana_datos, text=habitacion)
            label.pack()
        

 